hydra-cluster/src/CardanoNode.hs

```diff
@@ -125,6 +125,17 @@ getCardanoNodeVersion :: IO String
 getCardanoNodeVersion =
   readProcess "cardano-node" ["--version"] ""
 
+-- | Tries to reach for an existing cardano-node running on default state directory.
+findRunningCardanoNode :: FilePath -> KnownNetwork -> IO (Maybe RunningNode)
+findRunningCardanoNode stateDirectory knownNetwork = do
+  let knownNetworkId = toNetworId knownNetwork
+  n@RunningNode{networkId, nodeSocket} <- mkRunningNode stateDirectory defaultCardanoNodeArgs
+  guard (knownNetworkId == networkId)
```

Comment:
Must: not throw here, otherwise you would need to catch an exception around `findRunningCardanoNode`

`guard` in `IO` throws an exception if it's `False`, see https://hackage.haskell.org/package/base-4.19.0.0/docs/src/GHC.Base.html#line-1741

You _can_ do this, but then the signature `IO (Maybe RunningNode)` does not make sense and it should rather be `IO RunningNode`. Then you need to do a `try` or `catch` at the call site (and you can skip the `try` below)